tests/fullydistributed_grids/repartitioning_05.cc

```diff
@@ -0,0 +1,154 @@
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Test
+// TriangulationDescription::Utilities::create_description_from_triangulation()
+// so that it also works for p:d:T set up on a subcommunicator.
+
+#include <deal.II/base/mpi_consensus_algorithms.h>
+
+#include <deal.II/distributed/fully_distributed_tria.h>
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/fe/fe_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/tria_description.h>
+
+#include <deal.II/lac/la_parallel_vector.h>
+
+#include <deal.II/numerics/data_out.h>
+
+#include "./tests.h"
+
+using namespace dealii;
+
+MPI_Comm
+create_sub_comm(const MPI_Comm &comm, const unsigned int size)
+{
+  const auto rank = Utilities::MPI::this_mpi_process(comm);
+
+  int color = rank < size;
+
+  MPI_Comm sub_comm;
+  MPI_Comm_split(comm, color, rank, &sub_comm);
+
+  if (rank < size)
+    return sub_comm;
+  else
+    return MPI_COMM_NULL;
+}
+
+
+
+template <int dim, int spacedim>
+LinearAlgebra::distributed::Vector<double>
+partition_distributed_triangulation(const Triangulation<dim, spacedim> &tria_in,
+                                    const MPI_Comm &                    comm)
+{
+  const auto comm_tria = tria_in.get_communicator();
+
+  const auto n_global_active_cells = Utilities::MPI::max(
+    comm_tria == MPI_COMM_NULL ? 0 : tria_in.n_global_active_cells(), comm);
+
+  if (comm_tria == MPI_COMM_NULL)
+    {
+      LinearAlgebra::distributed::Vector<double> partition{
+        IndexSet(n_global_active_cells), IndexSet(n_global_active_cells), comm};
+      partition.update_ghost_values();
+      return partition;
+    }
+
+  const auto tria =
+    dynamic_cast<const parallel::TriangulationBase<dim, spacedim> *>(&tria_in);
+
+  Assert(tria, ExcNotImplemented());
+
+  LinearAlgebra::distributed::Vector<double> partition(
+    tria->global_active_cell_index_partitioner().lock()->locally_owned_range(),
+    tria->global_active_cell_index_partitioner().lock()->ghost_indices(),
+    comm);
+
+  const unsigned int n_partitions = Utilities::MPI::n_mpi_processes(comm);
+
+  for (const auto &cell : tria_in.active_cell_iterators())
+    if (cell->is_locally_owned())
+      partition[cell->global_active_cell_index()] =
+        cell->global_active_cell_index() * n_partitions /
+        tria_in.n_global_active_cells();
+
+  partition.update_ghost_values();
+
+  return partition;
+}
```

Comment:
@kronbichler This is how a partitioning policy might look like in this context. "Simply" replace the communicator in the partitioner...